# (Ford GoBike System Data)
## by (Reem Shaya)

## Preliminary Wrangling

> Ford GoBike is the Bay Area's bike share system, with thousands of bikes at hundreds of station in San Francisco, San Jose and the East Bay.

In [2]:
# import all packages and set plots to be embedded inline
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb

%matplotlib inline

> Load in your dataset and describe its properties through the questions below.
Try and motivate your exploration goals through this section.

In [4]:
df_ford = pd.read_csv('201902-fordgobike-tripdata.csv')
df_ford.head(50)

duration_sec                start_time                  end_time  \
0          52185  2019-02-28 17:32:10.1450  2019-03-01 08:01:55.9750   
1          42521  2019-02-28 18:53:21.7890  2019-03-01 06:42:03.0560   
2          61854  2019-02-28 12:13:13.2180  2019-03-01 05:24:08.1460   
3          36490  2019-02-28 17:54:26.0100  2019-03-01 04:02:36.8420   
4           1585  2019-02-28 23:54:18.5490  2019-03-01 00:20:44.0740   
5           1793  2019-02-28 23:49:58.6320  2019-03-01 00:19:51.7600   
6           1147  2019-02-28 23:55:35.1040  2019-03-01 00:14:42.5880   
7           1615  2019-02-28 23:41:06.7660  2019-03-01 00:08:02.7560   
8           1570  2019-02-28 23:41:48.7900  2019-03-01 00:07:59.7150   
9           1049  2019-02-28 23:49:47.6990  2019-03-01 00:07:17.0250   
10           458  2019-02-28 23:57:57.2110  2019-03-01 00:05:35.4350   
11           506  2019-02-28 23:56:55.5400  2019-03-01 00:05:21.7330   
12          1176  2019-02-28 23:45:12.6510  2019-03-01 00:04:49.1840   
13           915  2019-02-28 23:49:06.0620  2019-03-01 00:04:21.8670   
14           395  2019-02-28 23:56:26.8480  2019-03-01 00:03:01.9470   
15           208  2019-02-28 23:59:18.5480  2019-03-01 00:02:47.2280   
16           548  2019-02-28 23:50:41.6070  2019-02-28 23:59:49.9530   
17           674  2019-02-28 23:48:25.0950  2019-02-28 23:59:40.0920   
18           557  2019-02-28 23:49:01.8510  2019-02-28 23:58:19.8090   
19           874  2019-02-28 23:43:05.1830  2019-02-28 23:57:39.7960   
20           417  2019-02-28 23:50:38.2390  2019-02-28 23:57:35.8520   
21           414  2019-02-28 23:50:26.8790  2019-02-28 23:57:21.1300   
22           743  2019-02-28 23:44:56.4390  2019-02-28 23:57:20.2120   
23           367  2019-02-28 23:51:06.0140  2019-02-28 23:57:13.3120   
24           252  2019-02-28 23:52:51.1640  2019-02-28 23:57:03.9760   
25           360  2019-02-28 23:50:31.4310  2019-02-28 23:56:31.8910   
26           385  2019-02-28 23:49:24.3990  2019-02-28 23:55:50.2840   
27           408  2019-02-28 23:48:08.2820  2019-02-28 23:54:56.9300   
28           650  2019-02-28 23:43:27.5030  2019-02-28 23:54:18.4510   
29           629  2019-02-28 23:43:48.6580  2019-02-28 23:54:18.2540   
30           163  2019-02-28 23:50:45.6980  2019-02-28 23:53:29.5690   
31           223  2019-02-28 23:49:27.0270  2019-02-28 23:53:10.5350   
32           405  2019-02-28 23:45:39.2340  2019-02-28 23:52:24.8500   
33           638  2019-02-28 23:40:32.2730  2019-02-28 23:51:10.6390   
34           471  2019-02-28 23:42:43.3610  2019-02-28 23:50:34.4460   
35           318  2019-02-28 23:45:10.5550  2019-02-28 23:50:28.9050   
36           148  2019-02-28 23:47:02.9530  2019-02-28 23:49:31.1410   
37           305  2019-02-28 23:44:12.6760  2019-02-28 23:49:17.7000   
38          1066  2019-02-28 23:31:16.1390  2019-02-28 23:49:02.7920   
39           737  2019-02-28 23:35:45.4420  2019-02-28 23:48:03.1570   
40           328  2019-02-28 23:42:13.9100  2019-02-28 23:47:42.5430   
41           346  2019-02-28 23:40:56.0750  2019-02-28 23:46:42.4830   
42          1554  2019-02-28 23:20:24.0890  2019-02-28 23:46:19.0190   
43           116  2019-02-28 23:44:00.9880  2019-02-28 23:45:57.4820   
44           747  2019-02-28 23:32:59.4640  2019-02-28 23:45:26.9530   
45           717  2019-02-28 23:33:21.0950  2019-02-28 23:45:18.5100   
46           157  2019-02-28 23:41:35.8980  2019-02-28 23:44:12.9500   
47           417  2019-02-28 23:36:35.5790  2019-02-28 23:43:33.0270   
48           287  2019-02-28 23:38:32.5140  2019-02-28 23:43:20.4670   
49           203  2019-02-28 23:37:29.8330  2019-02-28 23:40:52.8700   

    start_station_id                                 start_station_name  \
0               21.0   Montgomery St BART Station (Market St at 2nd St)   
1               23.0                      The Embarcadero at Steuart St   
2               86.0                            Market St at Dolores St   
3              375.0           

In [8]:
df_ford.info()
print(df_ford.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 183412 entries, 0 to 183411
Data columns (total 16 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   duration_sec             183412 non-null  int64  
 1   start_time               183412 non-null  object 
 2   end_time                 183412 non-null  object 
 3   start_station_id         183215 non-null  float64
 4   start_station_name       183215 non-null  object 
 5   start_station_latitude   183412 non-null  float64
 6   start_station_longitude  183412 non-null  float64
 7   end_station_id           183215 non-null  float64
 8   end_station_name         183215 non-null  object 
 9   end_station_latitude     183412 non-null  float64
 10  end_station_longitude    183412 non-null  float64
 11  bike_id                  183412 non-null  int64  
 12  user_type                183412 non-null  object 
 13  member_birth_year        175147 non-null  float64
 14  memb

### What is the structure of your dataset?

There are (183412) bike rides in the dataset and 16 features :
duration_sec (int)
start_time (str)
end_time (str)
start_station_id (float)
start_station_name (str)
start_station_latitude (float)
start_station_longitude (float)
end_station_id (float)
end_station_name (str)
end_station_latitude (float)
end_station_longitude (float)
bike_id (int)
user_type (str)
member_birth_year (float)
member_gender (str)
bike_share_for_all_trip (str)

### What is/are the main feature(s) of interest in your dataset?

> This data is very exciting, and I think that all the features are important to me because they are complementary to each other and when they are interconnected together, I will see clear results and perceptions.. In any case, from my opinion, the most important features are the beginning and end of the journey, the year of birth and the gender of the user because this will explain to me the clients.

### What features in the dataset do you think will help support your investigation into your feature(s) of interest?

> Your answer here!

## Univariate Exploration

> In this section, investigate distributions of individual variables. If
you see unusual points or outliers, take a deeper look to clean things up
and prepare yourself to look at relationships between variables.

> Make sure that, after every plot or related series of plots, that you
include a Markdown cell with comments about what you observed, and what
you plan on investigating next.

### Discuss the distribution(s) of your variable(s) of interest. Were there any unusual points? Did you need to perform any transformations?

> Your answer here!

### Of the features you investigated, were there any unusual distributions? Did you perform any operations on the data to tidy, adjust, or change the form of the data? If so, why did you do this?

> Your answer here!

## Bivariate Exploration

> In this section, investigate relationships between pairs of variables in your
data. Make sure the variables that you cover here have been introduced in some
fashion in the previous section (univariate exploration).

### Talk about some of the relationships you observed in this part of the investigation. How did the feature(s) of interest vary with other features in the dataset?

> Your answer here!

### Did you observe any interesting relationships between the other features (not the main feature(s) of interest)?

> Your answer here!

## Multivariate Exploration

> Create plots of three or more variables to investigate your data even
further. Make sure that your investigations are justified, and follow from
your work in the previous sections.

### Talk about some of the relationships you observed in this part of the investigation. Were there features that strengthened each other in terms of looking at your feature(s) of interest?

> Your answer here!

### Were there any interesting or surprising interactions between features?

> Your answer here!

> At the end of your report, make sure that you export the notebook as an
html file from the `File > Download as... > HTML` menu. Make sure you keep
track of where the exported file goes, so you can put it in the same folder
as this notebook for project submission. Also, make sure you remove all of
the quote-formatted guide notes like this one before you finish your report!